File: src/main/java/mediathek/gui/actions/DeleteLocalFilmlistAction.java
package mediathek.gui.actions;

import mediathek.config.Konstanten;
import mediathek.config.StandardLocations;
import mediathek.mainwindow.MediathekGui;
import mediathek.tool.FileUtils;

import javax.swing.*;
import java.awt.event.ActionEvent;
import java.nio.file.Paths;

public class DeleteLocalFilmlistAction extends AbstractAction {
    private final JFrame owner;

    public DeleteLocalFilmlistAction(MediathekGui parent) {
        super();
        owner = parent;

        putValue(NAME, "Lokale Filmliste löschen");
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        var filmlistPathStr = StandardLocations.getFilmlistFilePathString();
        var fimlistPath = Paths.get(filmlistPathStr);
        FileUtils.moveToTrash(fimlistPath);
        JOptionPane.showMessageDialog(owner,
                "Filmliste wurde gelöscht.\nDas Programm wird nun beendet.",
                Konstanten.PROGRAMMNAME, JOptionPane.INFORMATION_MESSAGE);
        MediathekGui.ui().quitApplication();
    }
}

File: src/main/kotlin/mediathek/gui/actions/DeleteLocalFilmlistAction.kt
package mediathek.gui.actions

import mediathek.config.Konstanten
import mediathek.config.StandardLocations
import mediathek.mainwindow.MainWindowQuitHost
import mediathek.tool.FileUtils
import java.awt.event.ActionEvent
import java.nio.file.Paths
import javax.swing.AbstractAction
import javax.swing.JOptionPane

class DeleteLocalFilmlistAction(
    private val host: MainWindowQuitHost,
) : AbstractAction() {
    init {
        putValue(NAME, "Lokale Filmliste löschen")
    }

    override fun actionPerformed(event: ActionEvent?) {
        val filmlistPath = Paths.get(StandardLocations.getFilmlistFilePathString())
        FileUtils.moveToTrash(filmlistPath)
        JOptionPane.showMessageDialog(
            host.ownerFrame(),
            "Filmliste wurde gelöscht.\nDas Programm wird nun beendet.",
            Konstanten.PROGRAMMNAME,
            JOptionPane.INFORMATION_MESSAGE,
        )
        host.quitApplication()
    }
}
